### ISSUE TITLE
Sanitize ref's and engine

### ISSUE THREAD

**Issue by zgdump** [FWGS/xash3d-fwgs#847]
## Описание

В исследовательских целях собрался и запустился в режиме санитайзера. Ошибки сыпятся, как из ведра и в движке, и в рендерах. Хуже всего дела обстоят с `ref_soft`, ибо он не доходит даже до меню, падая в функции `GL_UploadTexture` файла `r_image.c`

## Как повторить

1. Configure:

```
.\waf configure -8 -T sanitize -o build/amd64-debug --prefix=build/amd64-debug-prefix --sdl2=C:\\SDL2 -P
``` 

2. Build:

```
.\waf install
```

3. Run:

```
.\build\amd64-debug-prefix\lib\xash3d\xash3d.exe -ref ... -dev 2 -log +map c1a0d
```

## Как ловить ошибки?

Во Windows использовал Visual Studio, который аттачил ко процессу сразу после запуска. Чтобы ловить ошибок больше, можно стартовать процесс сразу из IDE, но это не шибко удобно

## Примечание

На Windows, на последнем master, бектрейс пишется в консоль, которая моментально закрывается. Этой консоли будет посвящено отдельное issue, позже пропишу

**Comment by zgdump** [comment]
Не знаю, можно ли это issue превратить в описание всех найденных мест, которые нашёл санитайзер.

@0x4E69676874466F78 ещё до меня исследовал санитайзер, он же и подготовил патч для запуска оного под MSVC. Я же собирался исследовать `resman_texture`, но всё падает сильно раньше. на моделях

**Comment by a1batross** [comment]
Далеко не все из них критичны.

Но да, с sanitize я тоже запускаю довольно часто. Правда, жаль, что линковка санитайзеров очень много чего ломает -- например у меня перестает работать rpath, плюс надо pthread линковать всегда.

**Comment by zgdump** [comment]
> Далеко не все из них критичны.

Согласен. Но если собраться с ним, игра даже не стартует. Это не позволяет протестировать мой код
